subprojects/packagefiles/ftxui/meson.build

```diff
@@ -0,0 +1,96 @@
+project('ftxui', 'cpp',
+    version: '2.0.0',
+    meson_version: '>= 0.62.0',
+    default_options: ['cpp_std=c++17'],
+)
+
+if host_machine.system() == 'windows'
+  add_project_arguments('/utf-8', '-DUNICODE', language: 'cpp')
+endif
+
+# Upstream CMake supports only static linking,
+# so we do the same.
+
```

Comment:
This seems wrong. https://github.com/ArthurSonzogni/FTXUI/commit/57a5512a22aa1a9af570b932fbf8751b20039bdb

Reply:
Maybe it's just broken on MSVC? We do have `build_target(..., target_type: host_machine.system() == 'windows' ? 'static_library' : 'library')` for that...

Reply:
They don't have any `dllexport` so I guess that's why they can't build shared on Windows. Sounds like a job for https://github.com/mesonbuild/meson/pull/10199.

Reply:
That won't help in the slightest, it just creates a define that projects then have to care enough to use -- which they incredibly often do not. :(

Reply:
Anyway, I think it would be best to use the build_target() with target_type trick here so that at least linux and macOS users can defer to the builtin options.

Reply:
On the other hand I'm a big fan of simplicity. This has the same behaviour on all platforms and based on what we know of usage, most WrapDB users seem to prefer static linking.

Reply:
Hmm... the main reason I'd want to do regular library() for linux is to make it closer to upstream and thus more useful if attempting to submit as a PR to upstream, anyway.

That may not be super important here after all.

Reply:
> it just creates a define that projects then have to care enough to use

Ah, they indeed don't have a define at all yet, I was expecting they would have one for at least gnu visibility, or maybe using the cmake generator. Indeed that would need to change header files. Upstream could be interested to have it though.

> On the other hand I'm a big fan of simplicity. This has the same behaviour on all platforms and based on what we know of usage, most WrapDB users seem to prefer static linking.

I agree, if upstream does not deal with symbol visibility at all better not try doing shared library.

Reply:
> I agree, if upstream does not deal with symbol visibility at all better not try doing shared library.

TBH on Linux symbol visibility is *just* a performance optimization and prevents accidental use of non-public API. And the latter is just as much of a problem when linking statically.

The effort needed to bend over backward is probably a bigger concern.

Reply:
> the main reason I'd want to do regular library() for linux is to make it closer to upstream and thus more useful if attempting to submit as a PR to upstream, anyway.

That's a valid reason tbh. Maybe just use library() and add `if host_system == windows and default_library!=static error('not supported')` ? And link to an upstream bug report for adding the symbol visibility define.